src/planner/dishes/api.ts

```diff
@@ -0,0 +1,26 @@
+import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
+import { readCookie } from "../../common/cookieHelper";
+import { TDishesResponse } from "./types";
+
+const API_URL = process.env.REACT_APP_API_URL;
+
+export const dishesApi = createApi({
+  reducerPath: "dishesApi",
+  baseQuery: fetchBaseQuery({
+    baseUrl: `${API_URL}/api/dishes`,
+    prepareHeaders: (headers) => {
+        const token = readCookie("auth");
+        if (token) headers.set("Authorization", `Bearer ${token}`);
+        return headers;
+    }
+  }),
+  endpoints: (build) => ({
+    getDishes: build.query<TDishesResponse, void>({
+      query: () => {
+        return "";
+      },
```

Comment:
``` suggestion
      query: () => "",
```

Reply:
Are we not doing pagination or search functionality?

Reply:
I wanted to not be able to do complete copy paste in this PR so I can learn the language :p.

So I restricted myself to doing just a simple GetAll without all the other things like pagination and search. Can add all that in a separate PR.